src/main/java/org/openstreetmap/atlas/checks/validation/tag/TollValidationCheck.java

```diff
@@ -0,0 +1,601 @@
+        }
+        return false;
+    }
+
+    /**
+     * @param edge1
+     *            some edge
+     * @param edge2
+     *            some other edge
+     * @return boolean regarding if they have same highway tag?\
+     */
+    private boolean hasSameHighwayTag(final Edge edge1, final Edge edge2)
+    {
+        if (HighwayTag.highwayTag(edge1).isPresent() && HighwayTag.highwayTag(edge2).isPresent())
+        {
+            return edge1.highwayTag().equals(edge2.highwayTag());
+        }
+        return false;
+    }
+
+    /**
+     * @param tags
+     *            some edge tags
+     * @return if tags contains toll=yes
+     */
+    private boolean hasTollYesTag(final Map<String, String> tags)
+    {
+        return tags.keySet().stream().anyMatch(tag -> tag.equals(TollTag.KEY))
+                && tags.get(TollTag.KEY).equalsIgnoreCase(TollTag.YES.toString());
+    }
+
+    /**
+     * @param tags
+     *            some osm tags
+     * @return boolean for if the highway tag contains 'toll'
+     */
+    private boolean highwayTagContainsToll(final Map<String, String> tags)
+    {
+        return tags.get(HighwayTag.KEY).contains(TollTag.KEY);
+    }
+
+    /**
+     * @param inEdges
+     *            some inedges
+     * @param outEdges
+     *            some outedges
+     * @param edge
+     *            some edge
+     * @return boolean for inconsistent tagging
+     */
+    private boolean inconsistentTollTagLogic(final Set<Edge> inEdges, final Set<Edge> outEdges,
+            final Edge edge)
+    {
+        for (final Edge inEdge : inEdges)
+        {
+            for (final Edge outEdge : outEdges)
+            {
+                if (this.hasSameHighwayTag(edge, inEdge) && this.hasSameHighwayTag(edge, outEdge)
+                        && this.angleBetweenEdges(edge, outEdge) >= this.minAngleForContiguousWays
+                        && this.angleBetweenEdges(inEdge, edge) >= this.minAngleForContiguousWays)
+                {
+                    final Map<String, String> inEdgeOsmTags = inEdge.getOsmTags();
+                    final Map<String, String> outEdgeOsmTags = outEdge.getOsmTags();
+                    if (this.bothTollYesTag(inEdgeOsmTags, outEdgeOsmTags))
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param edgeInQuestion
+     *            the edge in question
+     * @param edgeInQuestionTags
+     *            tags of edge in question
+     * @return boolean if is case one
+     */
+    private boolean isCaseOne(final Edge edgeInQuestion,
+            final Map<String, String> edgeInQuestionTags)
+    {
+        return !this.hasTollYesTag(edgeInQuestionTags)
+                && !this.markedIntersectingTollFeature.contains(edgeInQuestion.getOsmIdentifier())
+                && this.edgeIntersectsTollFeature(edgeInQuestion);
+    }
+
+    /**
+     * @param edgeInQuestion
```

Comment:
It would be good to give a description of what this case covers here.

Reply:
Addressed and fixed!